packages/gitbook/src/components/Footer/Footer.tsx

```diff
@@ -17,125 +19,123 @@ export function Footer(props: {
-                            <p className={tcls('text-xs', 'text-dark/7', 'dark:text-light/6')}>
-                                {customization.footer.copyright}
-                            </p>
-                        ) : null}
+                        {partition(customization.footer.groups, FOOTER_COLUMNS).map(
+                            (column, columnIndex) => (
+                                <div key={columnIndex} className="flex w-full flex-col gap-10">
+                                    {column.map((group, groupIndex) => (
+                                        <FooterLinksGroup
+                                            key={groupIndex}
+                                            group={group}
+                                            context={context}
+                                        />
+                                    ))}
+                                </div>
+                            ),
+                        )}
                     </div>
-                    {customization.themes.toggeable ? (
-                        <div
-                            className={tcls(
-                                'flex',
-                                'flex-col',
-                                'items-start',
-                                'order-[-1]',
-                                'px-4',
-                                'sm:px-6',
-                                'md:px-0',
-                                'md:order-1',
-                            )}
-                        >
-                            <React.Suspense fallback={null}>
-                                <ThemeToggler />
-                            </React.Suspense>
-                        </div>
-                    ) : null}
+                )}
+
+                {/* Legal */}
+                <div
+                    className={tcls(
+                        'mx-auto w-full grow text-xs text-dark/7 dark:text-light/6 items-center text-center order-4 flex flex-col gap-2 empty:hidden',
+                        customization.footer.groups.length == 0 &&
+                            'order-2 lg:flex-1 lg:w-auto lg:items-start lg:max-w-3xl self-center lg:text-start',
+                        customization.footer.groups.length == 0 &&
+                            customization.footer.logo == undefined &&
+                            'sm:w-auto sm:flex-1 sm:items-start sm:max-w-3xl sm:text-start',
+                    )}
+                >
+                    {customization.footer.copyright && <p>{customization.footer.copyright}</p>}
                 </div>
             </div>
-        </div>
+        </footer>
+    );
+}
+
+function partition(array: any[], length: number) {
+    let rest = array.length % length;
+    let size = Math.floor(array.length / length);
+    let j = 0;
+
+    return Array.from({ length: Math.min(array.length, length) }, (_, i) =>
+        array.slice(j, (j += size + (i < rest ? 1 : 0))),
     );
```

Comment:
Maybe helpful to add some comments to make it a bit easier to read + put it in a utility folder somewhere, partitioning is a common thing that may be used elsewhere.